core/mutator/entry/entry.go

```diff
@@ -81,20 +76,20 @@ func (*Entry) Mutate(oldValue, mutation []byte) ([]byte, error) {
 		return nil, mutator.ErrMissingKey
 	}
 
-	if err := verifyKeys(oldValue, kv, update, entry); err != nil {
+	if err := VerifyKeys(oldValue, kv, update, entry); err != nil {
```

Comment:
Thanks. I started with exporting and calling the checks which happen in `Mutate`. It makes more sense to just call `Mutate` (which does the checks and return the new leaf which the monitor needs anyways).